airbyte-cdk/python/airbyte_cdk/sources/utils/stream_helpers.py

```diff
@@ -0,0 +1,44 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from typing import Any, Mapping, Optional
+
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.core import StreamData
+
+
+class StreamHelper:
+    def get_first_record(self, stream: Stream) -> StreamData:
+        """
+        Gets the first record for a stream.
+
+        :param stream: stream
+        :return: StreamData containing the first record in the stream
+        """
+        # Some streams need a stream slice to read records (e.g. if they have a SubstreamSlicer)
+        stream_slice = self.get_stream_slice(stream)
+        records = stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=stream_slice)
+        next(records)
```

Comment:
This part should be replaced with this to avoid `StopIteration` error on empty streams:

```python
class StreamHelper:
    def get_first_record(self, stream: Stream) -> StreamData:
        """
        Gets the first record for a stream.

        :param stream: stream
        :return: StreamData containing the first record in the stream
        """
        try:
            # Some streams need a stream slice to read records (e.g. if they have a SubstreamSlicer)
            stream_slice = self.get_stream_slice(stream)
            records = stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=stream_slice)
            return next(records)
        except StopIteration:
            return {}
```